File: Kvasha/1334.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @param {number} distanceThreshold
 * @return {number}
 */
let findTheCity = function (n, edges, distanceThreshold) {
  const MAX = 1e4 + 1;
  let D = Array.from({ length: n }, (_) => new Uint16Array(n).fill(MAX));
  let children = n - 1;
  let result = 0;

  for (let [u, v, w] of edges) {
    D[u][v] = D[v][u] = w;
  }

  for (let i = 0; i < n; ++i) {
    for (let u = 0; u < n; ++u) {
      if (D[u][i] === MAX) continue;

      for (let v = 0; v < n; ++v) {
        if (u === v) continue;

        D[u][v] = Math.min(D[u][v], D[u][i] + D[i][v]);
      }
    }
  }

  for (let i = 0; i < n; ++i) {
    let currChildren = 0;

    for (let j = 0; j < n; ++j) {
      currChildren += D[i][j] <= distanceThreshold;
    }

    if (currChildren <= children) {
      children = currChildren;
      result = i;
    }
  }

  return result;
};

File: 1334.py
class Solution:
    def findTheCity(self, n: int, edges: List[List[int]], distanceThreshold: int) -> int:

        previus_weights=[[100000] * n for i in range(n)]
        for i in range(n):
            previus_weights[i][i] = 0

        for edge in edges:
            previus_weights[edge[0]][edge[1]] = edge[2]
            previus_weights[edge[1]][edge[0]] = edge[2]

        for k in range(n):
            current_weights=[[100000] * n for i in range(n)]
            for i in range(n):
                current_weights[i][i] = 0
            for i in range(n):
                for j in range(n):
                    current_weights[i][j] = min(previus_weights[i][j], \
                        previus_weights[i][k] + previus_weights[k][j])
            previus_weights = current_weights
        
        result = 0
        neighbours_count = 10000000
        for i in range(n):
            count = 0
            for j in range(n):
                if previus_weights[i][j] <= distanceThreshold and i != j:
                    count+=1
            if count <= neighbours_count:
                result = i
                neighbours_count = count
        return result
